src/code/audio/code_801A5BD0.c

```diff
@@ -1,45 +1,780 @@
+        entry = &gSfxBanks[SFX_BANK(sfxId)][entryIndex];
+        if (entry->posX == &pos->x && entry->sfxId == sfxId) {
+            if (entry->state >= SFX_STATE_PLAYING_REFRESH) {
+                Audio_QueueCmdS8(0x06020000 | ((entry->channelIdx & 0xFF) << 8), 0);
+            }
+            if (entry->state != SFX_STATE_EMPTY) {
+                Audio_RemoveSfxBankEntry(SFX_BANK(sfxId), entryIndex);
+            }
+            entryIndex = 0xFF;
+        } else {
+            prevEntryIndex = entryIndex;
+        }
+
+        if (entryIndex != 0xFF) {
+            entryIndex = gSfxBanks[SFX_BANK(sfxId)][prevEntryIndex].next;
+        }
+    }
+
+    entryToRemove.posX = &pos->x;
+    entryToRemove.sfxId = sfxId;
+    Audio_RemoveMatchingSfxRequests(3, &entryToRemove);
+}
+
+void Audio_StopSfxByTokenAndId(u8 token, u16 sfxId) {
+    SfxBankEntry* entry;
+    u8 entryIndex = gSfxBanks[SFX_BANK(sfxId)][0].next;
+    u8 prevEntryIndex = 0;
+    SfxBankEntry entryToRemove;
+
+    while (entryIndex != 0xFF) {
+        entry = &gSfxBanks[SFX_BANK(sfxId)][entryIndex];
+        if (entry->token == token && entry->sfxId == sfxId) {
+            if (entry->state >= SFX_STATE_PLAYING_REFRESH) {
+                Audio_QueueCmdS8(0x06020000 | ((entry->channelIdx & 0xFF) << 8), 0);
+            }
+            if (entry->state != SFX_STATE_EMPTY) {
+                Audio_RemoveSfxBankEntry(SFX_BANK(sfxId), entryIndex);
+            }
+        } else {
+            prevEntryIndex = entryIndex;
+        }
+
+        if (entryIndex != 0xFF) {
+            entryIndex = gSfxBanks[SFX_BANK(sfxId)][prevEntryIndex].next;
+        }
+    }
+
+    entryToRemove.token = token;
+    entryToRemove.sfxId = sfxId;
+    Audio_RemoveMatchingSfxRequests(4, &entryToRemove);
+}
+
+void Audio_StopSfxById(u32 sfxId) {
+    SfxBankEntry* entry;
+    u8 entryIndex = gSfxBanks[SFX_BANK(sfxId)][0].next;
+    u8 prevEntryIndex = 0;
+    SfxBankEntry entryToRemove;
+
+    while (entryIndex != 0xFF) {
+        entry = &gSfxBanks[SFX_BANK(sfxId)][entryIndex];
+        if (entry->sfxId == sfxId) {
+            if (entry->state >= SFX_STATE_PLAYING_REFRESH) {
+                Audio_QueueCmdS8(0x06020000 | ((entry->channelIdx & 0xFF) << 8), 0);
+            }
+            if (entry->state != SFX_STATE_EMPTY) {
+                Audio_RemoveSfxBankEntry(SFX_BANK(sfxId), entryIndex);
+            }
+        } else {
+            prevEntryIndex = entryIndex;
+        }
+
+        entryIndex = gSfxBanks[SFX_BANK(sfxId)][prevEntryIndex].next;
+    }
+
+    entryToRemove.sfxId = sfxId;
+    Audio_RemoveMatchingSfxRequests(5, &entryToRemove);
+}
+
+void Audio_ProcessSfxRequests(void) {
+    if (gAudioContext.seqPlayers[SEQ_PLAYER_SFX].enabled) {
+        while (sSfxRequestWriteIndex != sSfxRequestReadIndex) {
+            Audio_ProcessSfxRequest();
+            sSfxRequestReadIndex++;
+        }
+    }
+}
+
+void Audio_SetUnusedBankLerp(u8 bankId, u8 target, u16 delay) {
```

Comment:
I'd prefer if we didn't call a function "Unused" unless there's literally no other way to distinguish it from a used one.